internal/network/machine_network_cidr.go

```diff
@@ -417,6 +450,57 @@ func GetInventoryNetworksByFamily(hosts []*models.Host, log logrus.FieldLogger)
 	return ret, nil
 }
 
+type CidrMetric struct {
+	cidr   string
+	metric int32
+}
+
+func GetDefaultRouteNetworkByFamily(hosts []*models.Host, log logrus.FieldLogger) map[AddressFamily]string {
+	ret := make(map[AddressFamily]string)
+	cidrmap := make(map[AddressFamily][]CidrMetric)
+	var err error
+
+	for _, h := range hosts {
```

Comment:
Maybe it is better to find first the selected route, and later find appropriate CIDR for it?

Reply:
Agree with Ori

Reply:
It can be done but really there is usually a single default route (or 2 at most). This map already contains only default routes and then I am sorting them according to metrics. The matching between the route and the CIDR was done above